File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mapping = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in mapping:
                return [mapping[complement],i]
            mapping[num] = i

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    //create an empty object to store numbers and their indices
    const map = {};
    // Iterate through the array
    for (let i = 0 ; i < nums.length; i++){
        const complement = target - nums[i];
        // Check if the complemet exist in our map
        if(map.hasOwnProperty(complement)){
            // If found, return the indices
            return [map[complement],i];
        }
        // Otherwise, store the current number with its index
        map[nums[i]] = i;
    }
};